components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -703,6 +685,51 @@ std::vector<uint8_t> HDKey::Sign(const std::vector<uint8_t>& msg, int* recid) {
   return sig;
 }
 
+std::vector<uint8_t> HDKey::SignBitcoin(base::span<const uint8_t, 32> msg) {
```

Comment:
Not sure now how to present error, added a TODO to fix that closer to a feature's complete stage

Reply:
I meant we can carry the current LOG(ERROR) message through `base::expected<std::vector<uint8_t>, std::string>` to caller and LOG the error message there or keep propagating. It would be better instead of checking empty vector to indicate failure. 
If we don't want to propagate error, we should at least use `absl::optional`.
I know it is the same old pattern in this file so that is why we should do it right on functions and then change old one later.

Reply:
wrapped in optional https://github.com/brave/brave-core/pull/17978/commits/2f117c534c8236292c2bbc209de5195830ccb9c5